test/reference/reference_test.go

```diff
@@ -68,13 +69,20 @@ func want() testobject.TestObject {
 	d.Parameter2 = 12
 	for i := 0; i < 22; i++ {
 		d.StructArray = append(d.StructArray, instantiations.InstantiatedTemplateStruct{
-			Parameter: d.Parameter2,
+			// field "Parameter" can remain unassigned, because it is
+			// implictly set due to the zserio definition
+			//Parameter: d.Parameter2,
```

Comment:
I guess whilst this would be set implicitly during serialization, we still need to add it in here in order to ensure that the go object represents what has been deserialized.